src/pkcs1v15/signing_key.rs

```diff
@@ -49,7 +49,7 @@ where
     }
 
     /// Generate a new signing key with a prefix for the digest `D`.
-    pub fn random<R: CryptoRngCore + ?Sized>(rng: &mut R, bit_size: usize) -> Result<Self> {
+    pub fn random<R: CryptoRngCore>(rng: &mut R, bit_size: usize) -> Result<Self> {
```

Comment:
Is there a particular reason you're getting rid of `?Sized` on these? It makes it possible to use a `dyn CryptoRngCore`

Reply:
> Is there a particular reason you're getting rid of ?Sized on these? It makes it possible to use a dyn CryptoRngCore

I don't think so, can revert probably

Reply:
I checked, and I removed it because I can't pass that constraint to `crypto_primes`

Reply:
Adding it to `crypto-primes` is unfortunately blocked by `crypto-bigint` not supporting it.

Reply:
Made https://github.com/RustCrypto/crypto-bigint/pull/760 for that